internal/rpcserver/interface.go

```diff
@@ -23,8 +26,26 @@ import (
 // The interface contract requires that all of these methods are safe for
 // concurrent access.
 type Peer interface {
-	// ToPeer returns the underlying peer instance.
-	ToPeer() *peer.Peer
+	// Addr returns the peer address.
+	Addr() string
```

Comment:
I don't really have any objection to these change here, but for posterity's sake, is there as a specific reason for it?

The `peer` package is not part of the main package, so it's acceptable to use the `peer.Peer` directly, just like it's fine to use `peer.StatsSnap` below.

That said, I haven't reviewed far enough to see the tests, so my suspicion is that it was done to be able to mock things that aren't exported from the `peer.Peer`.

Reply:
Yes, your suspicion is correct that I brought in this change to allow `peer.Peer` to be mocked (all fields in the `peer.Peer` struct are unexported). 

I do also think its a bit more consistent that with this change, `rpcserver` can access all peer related functions in the same manner without having to care if the underlying functionality exists in `serverPeer` or `peer.Peer` (versus currently having a few interface methods exposed for `serverPeer` functionality but then still calling `ToPeer()` everywhere for `peer.Peer` functionality). I don't have a strong opinion on this, though, and as mentioned, this was mainly brought in for testing purposes.